frontend/model/notifications/templates.js

```diff
@@ -1,133 +1,99 @@
-const dateNow = 1590823007327
-
-// TODO - add/improve logic to needed keys:
-// - avatar, name, linkTo, read
-export const MEMBER_ADDED = ({ timestamp, data }: TemplateArguments): TemplateDescription => {
-  const name = data.username // getDisplayName()
-  const avatar = ''
-  return {
-    avatar,
-    icon: 'user-plus',
-    level: 'info',
-    body: L('The group has a new member. Say hi to {name}!', {
-      name: `<strong>${name}</strong>`
-    }),
-    linkTo: '/group-chat/XXXX',
-    date: timeSince(timestamp, dateNow),
-    read: false
-  }
-}
-
-export const MEMBER_REMOVED = ({ timestamp, data }: TemplateArguments): TemplateDescription => {
-  const name = data.username
-  const avatar = ''
-  return {
-    avatar,
-    icon: 'user-minus',
-    level: 'danger',
-    // REVIEW @mmbotelho - Not only contributions, but also proposals.
-    body: L('{name} was kicked out of the group. Contributions were updated accordingly.', {
-      name: `<strong>${name}</strong>`
-    }),
-    linkTo: '/contributions',
-    date: timeSince(timestamp, dateNow),
-    read: true
-  }
-}
+type NewProposalType =
+  | 'ADD_MEMBER'
+  | 'CHANGE_MINCOME'
+  | 'CHANGE_VOTING_RULE'
+  | 'REMOVE_MEMBER';
 
-export const MEMBER_LEFT = ({ timestamp, data }: TemplateArguments): TemplateDescription => {
-  const name = data.username
-  const avatar = ''
-  return {
-    avatar,
-    icon: 'icon-minus',
-    level: 'info',
-    body: L('{name} has left your group. Contributions were updated accordingly.', {
-      name: `<strong>${name}</strong>`
-    }),
-    linkTo: '/contributions',
-    date: timeSince(timestamp, dateNow),
-    read: true
-  }
-}
-
-export const INCOME_DETAILS_OLD = ({ timestamp }: TemplateArguments): TemplateDescription => {
-  const avatar = ''
-  return {
-    avatar,
-    icon: 'coins',
-    level: 'info',
-    body: L("You haven't updated your income details in more than 6 months. Would you like to review them now?"),
-    linkTo: '/contributions/TODO-LINK-MODAL',
-    date: timeSince(timestamp, dateNow),
-    read: true
-  }
-}
+type NotificationLevel = 'danger' | 'info';
 
-export const SEND_CONTRIBUTION = ({ timestamp, data }: TemplateArguments): TemplateDescription => {
-  const month = 'June'
-  const avatar = ''
-  return {
-    avatar,
-    icon: 'coins',
-    level: 'info',
-    // REVIEW @mmbotelho - ranem contribution to payments?
-    body: L('Do not forget to send your {month} contributions.', {
-      month: `<strong>${month}</strong>`
-    }),
-    linkTo: '/payments',
-    date: timeSince(timestamp, dateNow),
-    read: true
-  }
+export type NotificationTemplate = {
+  +body: string;
+  +icon: string;
+  +level: NotificationLevel;
+  +linkTo: string;
+  +creator?: string;
+  +subtype?: string;
 }
 
-export const PROPOSAL_NEW = ({ timestamp, data }: TemplateArguments): TemplateDescription => {
-  const avatar = ''
-  const member = `<strong>${data.creator}</strong>`
+const strong = (text) => `<strong>${text}</strong>`
```

Comment:
This was already from before (this makes a more general implementation), but I wonder if it is a good idea to apply HTML like this. I'm unsure of what a good solution is when one wants markup inside a string that must be translated, but I'd start by escaping `&` with `&amp`, `>` with `&gt;` and `<` with '&lt;'.

Reply:
Good catch!